File: src/main/java/com/wikia/tibia/enums/City.java
package com.wikia.tibia.enums;

import com.fasterxml.jackson.annotation.JsonValue;

public enum City {

    AB_DENDRIEL("Ab'Dendriel"),
    ANKRAHMUN("Ankrahmun"),
    CARLIN("Carlin"),
    CORMAYA("Cormaya"),
    DARASHIA("Darashia"),
    DAWNPORT("Dawnport"),
    EDRON("Edron"),
    FARMINE("Farmine"),
    GRAY_BEACH("Gray Beach"),
    KAZORDOON("Kazordoon"),
    LIBERTY_BAY("Liberty Bay"),
    PORT_HOPE("Port Hope"),
    RATHLETON("Rathleton"),
    ROOKGAARD("Rookgaard"),
    SVARGROND("Svargrond"),
    THAIS("Thais"),
    VENORE("Venore"),
    YALAHAR("Yalahar");

    private String description;

    City(String description) {
        this.description = description;
    }

    @JsonValue
    public String getDescription() {
        return description;
    }
}

File: src/main/kotlin/com/wikia/tibia/enums/City.kt
package com.wikia.tibia.enums

import com.fasterxml.jackson.annotation.JsonValue
import com.wikia.tibia.interfaces.Description

enum class City(@JsonValue override val description: String) : Description {
  AB_DENDRIEL("Ab'Dendriel"),
  ANKRAHMUN("Ankrahmun"),
  CARLIN("Carlin"),
  CORMAYA("Cormaya"),
  DARASHIA("Darashia"),
  DAWNPORT("Dawnport"),
  EDRON("Edron"),
  FARMINE("Farmine"),
  FEYRIST("Feyrist"),
  GNOMEGATE("Gnomegate"),
  GRAY_BEACH("Gray Beach"),
  ISSAVI("Issavi"),
  ISLAND_OF_DESTINY("Island of Destiny"),
  KAZORDOON("Kazordoon"),
  LIBERTY_BAY("Liberty Bay"),
  MELUNA("Meluna"),
  NOSTALGIA("Nostalgia"),
  PORT_HOPE("Port Hope"),
  RATHLETON("Rathleton"),
  ROOKGAARD("Rookgaard"),
  ROSHAMUUL("Roshamuul"),
  SVARGROND("Svargrond"),
  THAIS("Thais"),
  TRAVORA("Travora"),
  VARIES("Varies"),
  VENORE("Venore"),
  YALAHAR("Yalahar"),
  EMPTY("");
}
